File: service/jvm-quarkus/src/main/java/com/ecobenchmark/entities/Task.java
package com.ecobenchmark.entities;

import org.hibernate.annotations.GenericGenerator;

import javax.persistence.*;
import java.time.Instant;
import java.util.UUID;

@Entity
public class Task {

    @Id
    @GeneratedValue(generator = "UUID")
    @GenericGenerator(name = "UUID", strategy = "org.hibernate.id.UUIDGenerator")
    private UUID id;

    private String name;

    private String description;

    @Column(name = "creation_date")
    private Instant creationDate;

    @ManyToOne
    private ListEntity list;

    public Task(String name, String description, Instant creationDate, ListEntity list) {
        this.name = name;
        this.description = description;
        this.creationDate = creationDate;
        this.list = list;
    }

    public Task() {
    }

    public UUID getId() {
        return id;
    }

    public void setId(UUID id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public Instant getCreationDate() {
        return creationDate;
    }

    public void setCreationDate(Instant creationDate) {
        this.creationDate = creationDate;
    }

    public ListEntity getList() {
        return list;
    }

    public void setList(ListEntity list) {
        this.list = list;
    }
}

File: service/jvm-kotlin-spring/kotlin-spring-jpa/src/main/kotlin/com/ecobenchmark/kotlinspringjpa/entities/Task.kt
package com.ecobenchmark.kotlinspringjpa.entities

import org.hibernate.annotations.GenericGenerator
import java.time.Instant
import java.util.*
import javax.persistence.Entity
import javax.persistence.GeneratedValue
import javax.persistence.Id
import javax.persistence.ManyToOne

@Entity
class Task(
    @ManyToOne
    var list: ListEntity,
    var name: String,
    var description: String,
    var creationDate: Instant,
) {
    @Id
    @GeneratedValue(generator = "UUID")
    @GenericGenerator(name = "UUID", strategy = "org.hibernate.id.UUIDGenerator")
    var id: UUID? = null
}
